File: c++/InterpreterPattern/SubExpression.cpp
#include "stdafx.h"
#include "SubExpression.h"


SubExpression::SubExpression(Expression *left,Expression *right) : OperatorExpression(left,right)
{

}


double SubExpression::Interpret(Context context)
{
	return this->left->Interpret(context) - this->right->Interpret(context);
}

File: csharp/DesignPattern.Interpreter/SubExpression.cs

using System.Collections.Generic;
namespace DesignPattern.Interpreter
{
    public class SubExpression : OperatorExpression
    {
        public SubExpression(Expression left, Expression right)
            : base(left, right)
        {

        }

        public override double Interpret(Context context)
        {
            return this.left.Interpret(context) - this.right.Interpret(context);
        }
    }
}
